fluss-flink/fluss-flink-common/src/main/java/com/alibaba/fluss/flink/source/enumerator/FlinkSourceEnumerator.java

```diff
@@ -242,29 +242,27 @@ public void start() {
                         this::checkPartitionChanges,
                         0,
                         scanPartitionDiscoveryIntervalMs);
+            } else if (streaming && !lakeEnabled && scanPartitionDiscoveryIntervalMs <= 0) {
+                // just call once
+                LOG.info(
+                        "Starting the FlussSourceEnumerator for table {} without partition discovery.",
+                        tablePath);
+                context.callAsync(this::listPartitions, this::checkPartitionChanges);
             } else {
-                if (!streaming) {
-                    startInBatchMode();
-                } else {
-                    // just call once
-                    LOG.info(
-                            "Starting the FlussSourceEnumerator for table {} without partition discovery.",
-                            tablePath);
-                    context.callAsync(this::listPartitions, this::checkPartitionChanges);
-                }
+                genHybridLakeAndFlussSplits();
```

Comment:
If lake enable + scanPartitionDiscoveryIntervalMs > 0 + streaming，
seems it won't discover new partitions? Right?
But we still need to discover new partitions.